### PR TITLE
Azure disk examples

### PR DESCRIPTION
**Pull request by andyzhangx**
**What this PR does / why we need it**:
This PR shows examples for mounting azure disk from v1.7.2, including managed disk support.

Original PR is https://github.com/kubernetes/kubernetes/pull/49468, per @sebgoa request, I have created this new PR. Original PR has already been reviewed by azure team.

### CODE REVIEW

**Comment by andyzhangx** [comment]
@ahmetb @jdumars for your awareness

File: staging/volumes/azure_disk/claim/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+
+
+## How does it work? 
+
+### Managed Disks
+The entire experience is offloaded to Azure to manage disks:storage accounts. You can use PVC (Kubernetes will automatically create a managed disk for you). Or you can you use an existing disk as PV in your PODs/RCs
+
+> as a general rule, use PV disks provisioned in the same Azure resource group where the cluster is provisioned.
```
Comment (ahmetb) [inline staging/volumes/azure_disk/claim/README.md]: add `> Note: ` here maybe?

File: staging/volumes/azure_disk/claim/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+> as a general rule, use PV disks provisioned in the same Azure resource group where the cluster is provisioned.   
+
+### Blob Disks 
+Blob Disks works in two modes. Controlled by #kind# parameter on the storage class. 
+
+### Dedicated (default mode)
+When *kind* parameter is set to *dedicated* K8S will create a new dedicated storage account for this new disk. No other disks will be allowed in the this storage account. The account will be removed when the PVC is removed (according to K8S PVC reclaim policy) 
+
+> You can still use existing VHDs, again the general rule apply use storage accounts that are part of cluster resource group
```
Comment (ahmetb) [inline staging/volumes/azure_disk/claim/README.md]: add `> Note: ` here maybe?

File: staging/volumes/azure_disk/claim/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+1. Maximum # of accounts created by K8S for shared PVC is **100**.
+2. Maximum # of disks per account is 60 (VHDs).
+3. K8S will create new account for new disks if * utilization(AccountType(NEWDISK)) > 50%  * keeping total # of accounts below 100.
+4. K8S will create initial 2 accounts ( 1 standard and 1 premium ) to accelerate the provisioning process.
+
+## Additional Notes
+The samples assumes that you have a cluster with node labeled with #disktype=blob# for VMs that are using blob disks and #disktype=managed# for VMs that are using managed disks. You can label your nodes or remove the node selector before using the files. 
+
+> You can not attach managed disks to VMs that are not using managed OS disks. This applies also the other way around no blob disks on VMS that are using managed OS disks
```
Comment (ahmetb) [inline staging/volumes/azure_disk/claim/README.md]: add `> Note: ` here maybe?

File: staging/volumes/azure_disk/claim/blob-based-disk/dedicated-hdd/pod-uses-dedicated-hdd.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  containers:
+  - image: nginx
+    name: az-c-01
+    command:
+    - "/bin/sh"
+    - "-c"
+    - while true; do echo $(date) >> /mnt/blobdisk/outfile; sleep 1; done
+    volumeMounts:
+    - name: blobdisk01
+      mountPath: "/mnt/blobdisk"
```
Comment (ahmetb) [inline staging/volumes/azure_disk/claim/blob-based-disk/dedicated-hdd/pod-uses-dedicated-hdd.yaml]: quote unnecessary (ditto other files)

File: staging/volumes/azure_disk/claim/blob-based-disk/dedicated-hdd/pod-uses-dedicated-hdd.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  labels:
+    name: storage
+spec:
+  containers:
+  - image: nginx
+    name: az-c-01
+    command:
+    - "/bin/sh"
+    - "-c"
+    - while true; do echo $(date) >> /mnt/blobdisk/outfile; sleep 1; done
```
Comment (ahmetb) [inline staging/volumes/azure_disk/claim/blob-based-disk/dedicated-hdd/pod-uses-dedicated-hdd.yaml]: seems inconsistent that you quoted "/bin/sh" "-c" and not this line (ditto other files)

File: staging/volumes/azure_disk/claim/README.md (hunk 1 above)
Comment (andyzhangx) [reply staging/volumes/azure_disk/claim/README.md]: fixed

File: staging/volumes/azure_disk/claim/README.md (hunk 2 above)
Comment (andyzhangx) [reply staging/volumes/azure_disk/claim/README.md]: fixed

File: staging/volumes/azure_disk/claim/README.md (hunk 3 above)
Comment (andyzhangx) [reply staging/volumes/azure_disk/claim/README.md]: fixed

File: staging/volumes/azure_disk/claim/blob-based-disk/dedicated-hdd/pod-uses-dedicated-hdd.yaml (hunk 5 above)
Comment (andyzhangx) [reply staging/volumes/azure_disk/claim/blob-based-disk/dedicated-hdd/pod-uses-dedicated-hdd.yaml]: I have removed all ", they are all unnecessary, thx for the check

File: staging/volumes/azure_disk/claim/blob-based-disk/dedicated-hdd/pod-uses-dedicated-hdd.yaml (hunk 4 above)
Comment (andyzhangx) [reply staging/volumes/azure_disk/claim/blob-based-disk/dedicated-hdd/pod-uses-dedicated-hdd.yaml]: fixed

**Comment by andyzhangx** [comment]
@ahmetb Thanks for the review, pls check again.

**Comment by ahmetb** [comment]
/lgtm

An additional reviewer from @kubernetes/sig-storage-pr-reviews would be great.

**Comment by andyzhangx** [comment]
@jdumars for review

**Comment by andyzhangx** [comment]
@brendandburns could you approve, it's been a long time, thx.

**Pull request merged**